docs/docs/examples/guides/05-jtc-guide.md

```diff
@@ -0,0 +1,587 @@
+that the robot should reach, while the latter refers to joint positions that have been recorded and are being published
+as named joint positions.
+
+In both cases, the length of `times_from_start` and `frames` or `joint_positions` need to be equal, such that every time
+corresponds to exactly one waypoint.
+
+For example, you could use:
+
+```yaml
+{
+  frames: [start, frame_1, frame_2, frame_3, start], 
+  times_from_start: [2.0, 4.0, 6.0, 8.0, 10.0]
+}
+```
+
+or
+
+```yaml
+{
+  joint_positions: [start, jconfig_1, jconfig_2, jconfig_3, start], 
+  times_from_start: [2.0, 4.0, 6.0, 8.0, 10.0]
+}
+```
+
+In the following section, we will demonstrate how these frames or joint positions can be easily recorded through the
+`3D Viz` and used with JTC in a matter of clicks.
+
+## Putting an application together
+
+Let us now go back to the application we created earlier in the guide, that consists of a hardware interface with the
+`Generic six-axis robot arm` and a `Joint Trajectory Controller`.
+
+
+### Recording frames and joint positions
+
+#### Create a frame from scratch
+
+1. Play the application and click on the `3D Viz` button. On the top-left corner you will see various visualization
+options.
+2. Expand the `Create frame` option and give the name `frame_1` before pressing `Create`.
+3. A frame will appear close to the origin of the map. You can click on the axes of the frame and drag it anywhere in
+the scene. Also, mind that this is a desired tool for your robot's end effector, therefore, you likely have to rotate it
+accordingly such that it's reachable both in terms of distance and orientation. You may also expand the `View settings`
+and toggle on `Show frames`, as this will provide a visual reference about the orientation of the end effector. For
+example:
+
+<div class="text--center">
+  <img src={jtcGuideCreateFrame} alt="Creating a frame in AICA Studio" />
+</div>
+
+Open your YAML editor and notice that there is now a `frames` section with your recorded frame, for example:
+
+```yaml
+frames:
+  frame_1:
+    reference_frame: world
+    position:
+      x: 0.468317
+      y: -0.116919
+      z: 0.422861
+    orientation:
+      w: 0.057171
+      x: 0.64853
+      y: 0.75827
+      z: 0.034142
+```
+
+#### Record a frame: Create a frame from an existing one 
+
+Alternatively, you can record a frame directly on the robot's end effector and then proceed to move it (in which case
+the orientation is copied from the end effector), as shown below:
+
+<div class="text--center">
+  <img src={jtcGuideRecordFrame} alt="Recording a frame in AICA Studio" />
+</div>
+
+```yaml
+frames:
+  frame_1:
+    reference_frame: world
+    position:
+      x: 0.493899
+      y: 0.140303
+      z: 0.429977
+    orientation:
+      w: -0.000563
+      x: 0.707388
+      y: 0.706825
+      z: 0.000001
+```
+
+You may also try the inverse, that is, copy one of the above code blocks and paste it right below the `dependencies`
```

Comment:
Find that quite specific, you can technically add it anywhere

Reply:
true, but I do want to guide them enough to not paste it under components or something. We could go "Paste it anywhere in the outermost scope" but I don't know if that's also very user-friendly

Reply:
fair point, yet `below the dependencies` is about as user friendly as "top level" because the same error could happen in any case

Reply:
yup, fair enough. I'll try to rephrase a bit there